controllers/vdoconfig_controller.go

```diff
@@ -901,44 +1000,177 @@ func (r *VDOConfigReconciler) reconcileCSISecret(ctx vdocontext.VDOContext, conf
 	return config, nil
 }
 
-func (r *VDOConfigReconciler) compareVersions(currentVersion, latestVersion string) (version string) {
+// compareVersions checks if the given version lies between min and max version
+func (r *VDOConfigReconciler) compareVersions(minVersion, currentVersion, maxVersion string) (bool, error) {
+	minVersionList := strings.Split(minVersion, ".")
 	currentVersionList := strings.Split(currentVersion, ".")
-	latestVersionList := strings.Split(latestVersion, ".")
-	for i := range currentVersionList {
-		version1, _ := strconv.Atoi(currentVersionList[i])
-		version2, _ := strconv.Atoi(latestVersionList[i])
-		if version1 > version2 {
-			latestVersion = currentVersion
+	maxVersionList := strings.Split(maxVersion, ".")
+
+	var isCompatibleVersion bool
+
+	for x := range minVersionList {
+		if (strings.Contains(minVersionList[x], "x")) || (strings.Contains(minVersionList[x], "X")) {
+			continue
+		}
+
+		version1, err := strconv.Atoi(minVersionList[x])
+
+		if err != nil {
+			return false, err
+		}
+		version2, _ := strconv.Atoi(currentVersionList[x])
+
+		if version2 > version1 {
+			isCompatibleVersion = true
 			break
+		} else if version1 == version2 {
+			isCompatibleVersion = true
+		} else {
+			isCompatibleVersion = false
 		}
 	}
-	return latestVersion
+
+	if isCompatibleVersion {
+		isCompatibleVersion = false
+		for x := range maxVersionList {
+			version1, _ := strconv.Atoi(currentVersionList[x])
+
+			if (strings.Contains(maxVersionList[x], "x")) || (strings.Contains(maxVersionList[x], "X")) {
+				continue
+			}
+
+			version2, err := strconv.Atoi(maxVersionList[x])
+			if err != nil {
+				return false, err
+			}
+
+			if version2 > version1 {
+				isCompatibleVersion = true
+				break
+			} else if version1 == version2 {
+				isCompatibleVersion = true
+			} else {
+				isCompatibleVersion = false
+			}
+		}
+	}
+
+	return isCompatibleVersion, nil
 }
 
-func (r *VDOConfigReconciler) fetchCsiDeploymentYamls(matrix CompatMatrix) (deploymentYamls []string) {
-	// Currently fetching the latest version of CSI and corresponding Deployment Yamls
-	// TODO Fetch vsphere and k8s version from the cluster and select suitable CSI version
+// compareSkewVersions checks if the given version matches with the skew version
+func (r *VDOConfigReconciler) compareSkewVersions(currentVersion, supportedVersion string) (bool, error) {
+
+	currentVersionList := strings.Split(currentVersion, ".")
+	supportedVersionList := strings.Split(supportedVersion, ".")
+
+	for x := range supportedVersionList {
+		if (strings.Contains(supportedVersionList[x], "x")) || (strings.Contains(supportedVersionList[x], "X")) {
```

Comment:
please explore using https://github.com/hashicorp/go-version for version comparisons